### PR TITLE
Auto-Instrumented Mongo and http server example

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
Adds a basic http server that uses Mongoose.

### CODE REVIEW

File: signalfx-tracing/signalfx-nodejs-tracing/mongo/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+[SignalFx Tracing Library for JavaScript](https://github.com/signalfx/signalfx-nodejs-tracing).
+Please examine the instrumented [client](./client.js) and [server](./server.js) for
+some basic patterns in accessing the instrumentations of a [Mongoose](http://mongoosejs.com/)
+client (via [mongodb-core](https://github.com/mongodb-js/mongodb-core)) and a basic
+[http](https://nodejs.org/api/http.html) server. This example is of a simple
+logging system that is auto-instrumented by a lone [tracer invocation](./logger/tracer.js).
+
+## Building the example app and client
+
+To run this example locally and send traces to your available Smart Agent or Gateway,
```
Comment (sm7x) [inline signalfx-tracing/signalfx-nodejs-tracing/mongo/README.md]: Instruct users to clone the repo

**Comment by keitwb** [comment]
I find the nesting under `signalfx-tracing` dir a bit confusing, like it makes it harder to find things.  Seems like both the express and mongo examples could just go under a common `nodejs` directory in the root of the repo.

File: signalfx-tracing/signalfx-nodejs-tracing/mongo/server.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      return res.end()
+    }
+    const response = {}
+    const activeSpan = scope.active()
+    if (req.method === 'GET') {
+      // Manually create a span for the GET request handler and make it a child
+      // of the automatically created http server's request span.
+      const handleGetSpan = tracer.startSpan('handleGet', { childOf: activeSpan })
+      // Activate the span via the Scope for the subsequent anonymous function.
+      return scope.activate(handleGetSpan, () => {
```
Comment (keitwb) [inline signalfx-tracing/signalfx-nodejs-tracing/mongo/server.js]: Does this enable the scope for the callback passed to the `handleGet` function too?

File: signalfx-tracing/signalfx-nodejs-tracing/mongo/logger/client.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      console.log('Goodbye!')
+    })
+
+    rl.on('line', (input) => {
+      if (!input) {
+        return rl.prompt()
+      } else if (input === '/q') {
+        return rl.close()
+      }
+      request({ method: 'POST', uri: config.serverUrl, json: { id, input } },
```
Comment (keitwb) [inline signalfx-tracing/signalfx-nodejs-tracing/mongo/logger/client.js]: Maybe show some manipulation of the auto-instrumented http client span if possible.

**Comment by rmfitzpatrick** [comment]
> I find the nesting under `signalfx-tracing` dir a bit confusing, like it makes it harder to find things. Seems like both the express and mongo examples could just go under a common `nodejs` directory in the root of the repo.

@keitwb, until recently the stated purpose of the tracing examples has been manual instrumentation w/ 3rd party tracers, so having a new parent `signalfx-tracing` directory for each offered library seems more reasonable than combining the approaches: https://github.com/signalfx/tracing-examples#auto-instrumentation

In the future I could see having two root directories, one for manual and one for signalfx-tracing.

File: signalfx-tracing/signalfx-nodejs-tracing/mongo/server.js (hunk 2 above)
Comment (rmfitzpatrick) [reply signalfx-tracing/signalfx-nodejs-tracing/mongo/server.js]: Yes, though this needs thorough testing as I've seen a few cases where this isn't functional: https://github.com/signalfx/signalfx-nodejs-tracing/blob/master/docs/API.md#scopeactivatespan-fn.  Native promises should have the correct active span too.

File: signalfx-tracing/signalfx-nodejs-tracing/mongo/README.md (hunk 1 above)
Comment (rmfitzpatrick) [reply signalfx-tracing/signalfx-nodejs-tracing/mongo/README.md]: updated

File: signalfx-tracing/signalfx-nodejs-tracing/mongo/logger/client.js (hunk 3 above)
Comment (rmfitzpatrick) [reply signalfx-tracing/signalfx-nodejs-tracing/mongo/logger/client.js]: Not able to intuitively do it w/ request (would need to add response event handlers), so I added parent spans to accomplish this.

**Pull request merged**